src/components/application_manager/test/commands/mobile/scrollable_message_request_test.cc

```diff
@@ -0,0 +1,261 @@
+#include "application_manager/event_engine/event.h"
+#include "mobile/scrollable_message_request.h"
+#include "interfaces/MOBILE_API.h"
+#include "application_manager/policies/policy_handler_interface.h"
+
+namespace test {
+namespace components {
+namespace commands_test {
+namespace mobile_commands_test {
+namespace scrollable_message_request {
+
+using ::testing::_;
+using ::testing::Return;
+using ::testing::ReturnRef;
+using am::commands::ScrollableMessageRequest;
+using am::commands::MessageSharedPtr;
+using am::event_engine::Event;
+using am::MockMessageHelper;
+
+namespace mobile_result = mobile_apis::Result;
+namespace am = ::application_manager;
+namespace hmi_response = am::hmi_response;
+
+typedef SharedPtr<ScrollableMessageRequest> CommandPtr;
+
+namespace {
+const uint32_t kTimeOut = 30000u;
+}
+
+typedef NiceMock<
+    ::test::components::application_manager_test::MockHMICapabilities>
+    MockHMICapabilities;
+
+class ScrollableMessageRequestTest
+    : public CommandRequestTest<CommandsTestMocks::kIsNice> {
+ public:
+  ScrollableMessageRequestTest() : app_(CreateMockApp()) {
+    msg_ = CreateMessage(smart_objects::SmartType_Map);
+  }
+
+ protected:
+  MockAppPtr app_;
+  MessageSharedPtr msg_;
+};
+
+TEST_F(ScrollableMessageRequestTest, Init_CorrectTimeout_SUCCESS) {
+  CommandPtr command(CreateCommand<ScrollableMessageRequest>(msg_));
+  (*msg_)[am::strings::msg_params][am::strings::timeout] = kTimeOut;
+  (*msg_)[am::strings::msg_params][am::strings::interaction_mode] =
+      mobile_apis::InteractionMode::MANUAL_ONLY;
+  EXPECT_EQ(kDefaultTimeout_, command->default_timeout());
+  command->Init();
+  EXPECT_EQ(kTimeOut, command->default_timeout());
+}
+
+TEST_F(ScrollableMessageRequestTest, Init_CorrectTimeout_UNSUCCESS) {
+  CommandPtr command(CreateCommand<ScrollableMessageRequest>(msg_));
+  (*msg_)[am::strings::msg_params][am::strings::interaction_mode] =
+      mobile_apis::InteractionMode::MANUAL_ONLY;
+  EXPECT_EQ(kDefaultTimeout_, command->default_timeout());
+  command->Init();
+  EXPECT_EQ(kTimeOut, command->default_timeout());
+}
+
+TEST_F(ScrollableMessageRequestTest, Run_ApplicationIsNotRegistered_UNSUCCESS) {
+  CommandPtr command(CreateCommand<ScrollableMessageRequest>(msg_));
+
+  ON_CALL(mock_app_manager_, application(_))
+      .WillByDefault(Return(ApplicationSharedPtr()));
+
+  EXPECT_CALL(
+      mock_app_manager_,
+      ManageMobileCommand(
+          MobileResultCodeIs(mobile_result::APPLICATION_NOT_REGISTERED), _));
+
+  command->Run();
+}
+
+TEST_F(ScrollableMessageRequestTest, Run_ProcessingResult_UNSUCCESS) {
+  (*msg_)[am::strings::msg_params][am::strings::soft_buttons] = 0;
+  CommandPtr command(CreateCommand<ScrollableMessageRequest>(msg_));
+
+  ON_CALL(mock_app_manager_, application(_)).WillByDefault(Return(app_));
+
+  ON_CALL(mock_app_manager_, GetPolicyHandler())
+      .WillByDefault(
+          ReturnRef(*static_cast<policy::PolicyHandlerInterface*>(NULL)));
```

Comment:
@indlu  _A NULL pointer points to regions filled with dragons, demons, core dumps, and numberless other foul creatures, all of which delight in frolicing in thy program if thou disturb their sleep._ Please fix this.